src/overlays/actors/ovl_Obj_Boat/z_obj_boat.c

```diff
@@ -28,31 +32,164 @@ const ActorInit Obj_Boat_InitVars = {
+        this->unk15D = 1;
+        this->dyna.actor.world.pos.x = this->unk164[this->unk15C].x;
+        this->dyna.actor.world.pos.z = this->unk164[this->unk15C].z;
+        this->dyna.actor.shape.rot.y = func_80B9AF50(this, &sp24);
+        this->dyna.actor.world.rot.y = this->dyna.actor.shape.rot.y;
+        this->unk15D = -this->unk15D;
+    }
+}
+
+void ObjBoat_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ObjBoat* this = THIS;
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+}
+
+void func_80B9B124(ObjBoat* this) {
+    this->unk160 += 0x3E8;
+    this->dyna.actor.world.pos.y = Math_SinS(this->unk160) + this->dyna.actor.home.pos.y;
+    this->dyna.actor.shape.rot.x = (s16)(s32)(Math_SinS(this->unk160) * 100.0f);
+    this->dyna.actor.shape.rot.z = (s16)(s32)(Math_SinS((s16)(this->unk160 * 2)) * 50.0f);
+}
+
+void ObjBoat_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    ObjBoat* this = THIS;
+    Player* player;
+    s32 currentAction;
+    f32 sp3C;
+    s16 sp3A;
+    Vec3f sp2C;
+
+    player = GET_PLAYER(globalCtx);
+    currentAction = DynaPolyActor_IsInRidingMovingState(&this->dyna);
+    sp3C = 0.0f;
+    sp3A = this->dyna.actor.shape.rot.y;
+    if ((currentAction) || ((DynaPolyActor_IsInRidingFallingState(&this->dyna)))) {
+        if ((this->unk15F == 0) &&
+            (((this->dyna.actor.params & 0x4000)) || ((currentAction != 0) && (this->unk15C == this->unk15E)))) {
+            this->unk15D = -this->unk15D;
+            if (this->unk15D > 0) {
+                this->unk15E = this->unk163;
+            } else {
+                this->unk15E = 0;
+            }
+            this->unk15F = 0x3C;
+        }
+    } else if (this->dyna.actor.speedXZ == 0.0f) {
+        if (this->unk15F != 0) {
+            this->unk15F--;
+        }
+    }
+    if (this->unk15C != this->unk15E) {
+        sp3A = func_80B9AF50(this, &sp2C);
+        if (Math_Vec3f_DistXZ(&this->dyna.actor.world.pos, &sp2C) < 200.0f) {
+            this->unk15C += this->unk15D;
+            if (this->unk15C == this->unk15E) {
+                if ((this->dyna.actor.params & 0x4000)) {
+                    this->unk15C = 0;
+                } else if (this->dyna.actor.speedXZ == 0.0f) {
+                    this->unk15C = 0;
+                    this->unk15D = -1;
+                }
+            }
+        } else {
+            sp3C = this->unk15D * ((this->dyna.actor.params & 0x4000) ? 5.0f : 3.0f);
+        }
+    }
+    if (player->csMode != 0x1A) {
+        Math_ScaledStepToS(&this->dyna.actor.shape.rot.y, sp3A, (s16)(s32)(fabsf(this->dyna.actor.speedXZ) * 40.0f));
+        this->dyna.actor.world.rot.y = this->dyna.actor.shape.rot.y;
+        Math_StepToF(&this->dyna.actor.speedXZ, sp3C, 0.05f);
+        Actor_MoveWithGravity(&this->dyna.actor);
+        if (this->dyna.actor.speedXZ != 0.0f) {
+            func_800B9010(&this->dyna.actor, 0x2154U);
```

Comment:
```suggestion
            func_800B9010(&this->dyna.actor, NA_SE_EV_PIRATE_SHIP - SFX_FLAG);
```
Can replace this with the correct sfx define